### PR TITLE
Support for a config file

### LINKED ISSUE #47: Introduce configuration file
Right now it is super annoying to always put the tokens, AI model names, custom prompt etc (the flags) to each and every source and provider.
We want to support a configuration file.
The following (generic) format is possible:
```json
{
  "github": {
    "token": "abc",
    "issue": "owner/repo/1"
  },
  "google": {
    "token": "abc",
    "model": "someModel"
  },
  "openai": {
    "token": "env:OPENAI_API_KEY",
    "model": "gpt-4"
  }
}
```

It seems that using with `Viper` this is fairly easy to implement.
We can also support not only json but also different configuration languages.

We also want to support `env:ABC` as keys and read from env variables in that case.
This is a security feature for tokens that we don't want to add to a plain text file.

The configuration file can be placed in various places:
1. `./config.json`                    // Current directory
2. `$HOME/.config/appname/config.json` // XDG Base Directory
3. `$HOME/.appname/config.json`       // Legacy/simple approach

(or what Viper gives us out of the box 🤷 )

Basically the configuration can contain each and every `command` including each and every `flag` as configuration.

The config file should be "dynamically".
In case something is NOT defined, we go with the "default behaviour" on cmd (Cobra) level.

The following should give presence from top to bottom)
* CLI flag
* Config value
* CLI default
* CLI throws

### PR DESCRIPTION
**Pull request by StefMa**
fixes #47

### CODE REVIEW

**Comment by StefMa** [comment]
I probably went a bit too far with that 🙈
I thought it would be nice to have kind of ever input as a possible config. However I guess this violance a bit of that concept.

So what would you recommend? Make it possible to only configure the tokens and that's it? 🤔

**Comment by rsj-ioki** [comment]
IMO you don't really to add support for config files. For the simplicity of this app, everything via CLI should be just fine.

**Comment by StefMa** [comment]
Well, "the problem" is, that I use multiple AI providers. Each of them needs another token. Right now I always have to open my password manager, search for the provider and copy paste the token into it.
That's pretty annoying.
So I thought I can simplify this by using a config file where I store the tokens so I don't have to do that process each time again and again...
(ignoring the security problem that I would store api tokens in plain text in disk) 

Would you recommend another approach to avoid this? 🤔

**Comment by StefMa** [comment]
So..? Only tokens in the config file? 🙃

**Comment by StefMa** [comment]
@rsj-ioki can you please do a check if this is fine now 🙃 
I removed the "always dynamic" stuff from the config file while keeping the more static-like things.
Like AI models, URLs and tokens.

Please also review the code 🙏 
Thanks!

File: internal/cli/cli.go [hunk 1]
```diff
@@ -45,14 +46,19 @@ summaraizer github --issue ioki-mobility/summaraizer/1 --token <token>`,
...
 	cmd.Flags().String(flagToken, "", "The GitHub token. Only required for private repositories.")
+	if tokenConfig := c.GetString("github.token"); tokenConfig != "" {
+		cmd.Flags().Set(flagToken, tokenConfig)
+	}
 
 	return cmd
 }
 
-func redditCmd() *cobra.Command {
+func redditCmd(c *config) *cobra.Command {
```
Comment (rsj-ioki) [inline internal/cli/cli.go:61]: Remove ununsed param. Add it back when you actually need it.

File: internal/cli/config.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		envWithoutPrefix := strings.TrimPrefix(s, "SUMMARAIZER_")
+		return strings.Replace(strings.ToLower(envWithoutPrefix), "_", ".", -1)
+	}), nil)
+	return &config{
+		k: k,
+	}
+}
+
+func (c *config) GetString(path string) string {
+	return c.k.String(path)
```
Comment (rsj-ioki) [inline internal/cli/config.go:32]: Here is some hint on how to leverage Go types to create a well organized config. No need to use error prone string keys. Plus you get some basic type checking for free. You also get to define some defaults in a nice way. You will need a bit more work on below snippet but you get the idea. So for now, current use of koanf package is okay but later try improving like below.

```diff
+type Config struct {
+	Slack  SlackConfig
+	Ollama OllamaConfig
+}
+
+type SlackConfig struct {
+	Token secret.Text
+}
+
+type OllamaConfig struct {
+	URL    string
+	Model  string
+	Prompt string
+}
+
+fun Load() *Config {
+	// add some default values if you like
+	cfg := &Config{
+		Ollama: OllamaConfig{
+			Prompt: "some default prompt",
+		},
+	}
+
+	ko := koanf.New("")
+	ko.Unmarshal("path/to/config", cfg)
+
+	return cfg
+}

```

Comment (StefMa) [reply internal/cli/config.go]: Done here https://github.com/ioki-mobility/summaraizer/pull/48/commits/db4448212cd2811e630f565d1b85bc08175decbb
Thanks fort he tip!

**Pull request merged**